.github/workflows/scan-images.yaml

```diff
@@ -30,41 +30,54 @@ jobs:
           remove-android: 'true'
           remove-codeql: 'true'
           remove-docker-images: 'true'
+
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+
       - name: Setup tools
         id: setup
         run: |
           sudo snap install yq
           echo "date=$(date '+%Y-%m-%d-%H-%M-%S')" >> $GITHUB_OUTPUT
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
```

Comment:
Ah, bad copy paste. Removing the `${{ matrix.python-version }}`.

Regarding the Python version, there's no hard dependency in 3.12. I needed a newer version from what Ubuntu 20.04 had because I was getting the following error:
```
Traceback (most recent call last):
  File "scripts/airgapped/get-all-images.py", line 66, in <module>
    def get_analytics_app_images(app: dict) -> list[str]:
TypeError: 'type' object is not subscriptable
```

Then I bumped the Ubuntu version to 24.04, but this had the issue that in that version I can't install pip packages in the root python installation. So I ended up using the setup python action, that sets up a virtual environment.

Thus the updates of Ubuntu and this python action. I've hardcoded to 3.10 for now but let me know if you have something else in mind

Reply:
https://github.com/canonical/bundle-kubeflow/pull/1038/commits/3ab0a2a6eeb863aadcb3795bd138514ebb9617bf

Reply:
> Regarding the Python version, there's no hard dependency in 3.12. I needed a newer version from what Ubuntu 20.04 had because I was getting the following error:

gotcha, I guess you could've tried using `typing.List` in py3.8, though since this version of Python is going EOL soon and `List` was deprecated in 3.9, we don't have to do anything about it.

> Then I bumped the Ubuntu version to 24.04, but this had the issue that in that version I can't install pip packages in the root python installation. So I ended up using the setup python action, that sets up a virtual environment.

Do you mean `pip3 install -r scripts/airgapped/requirements.txt` was giving you trouble? What kind of issue did you encountered?

Reply:
> Do you mean `pip3 install -r scripts/airgapped/requirements.txt` was giving you trouble? What kind of issue did you encountered?

Yes, it was something along the lines of https://stackoverflow.com/questions/77676556/whats-the-correct-way-to-use-user-local-python-environment-under-pep668

Reply:
In all our CIs this has not been an issue, so I'm unsure what the actual issue could be. I think that we can resolve this thread if you:

* Add a comment saying why we are pinning to this version of python and when it's recommended to update/remove it. Ideally you'd link here an issue where we are tracking that.
* Create an issue/enhancement proposal to get rid of the pyenv and the pinning

Reply:
But all the CI is using Ubuntu 20.04. The issue happens in 24.04, whose Python includes [PEP668](https://peps.python.org/pep-0668/), thus we didn't see it in the rest of the CI.

I can create an issue though to mention that we can't bump our CIs to 24.04 unless we use virtual environments

> Create an issue/enhancement proposal to get rid of the pyenv and the pinning

It's the other way around we will have to use venv everywhere because we won't be able to install packages in the system-wide python (we could, but they don't recommend it thus the above PEP)

Reply:
Actually, I ended up doing the following:
1. Used Ubuntu **22.04** (whose python doesn't have [PEP668](https://peps.python.org/pep-0668/))
2. Removed the python pinning https://github.com/canonical/bundle-kubeflow/pull/1038/commits/a31ae34ae6ca10777d380d1400c889bb4477d536
3. The tests will install everything in the global python installation (which will break in 24.04 and onwards)

So for now I'll stick with 22.04, but have created an issue to track why we can't go to 24.04 in https://github.com/canonical/bundle-kubeflow/issues/1057

Reply:
@DnPlas I'll resolve this conversation now since we have the issue to document the potential future issue with 24.04 and the pinning is removed in this PR